### PR TITLE
Compatibility of Mark GMail read with TB 68+

### PR DESCRIPTION
**Pull request by peci1**
Not giving up!

### CODE REVIEW

**Comment by tooomm** [comment]
CC @Qrzysio

**Comment by Qrzysio** [comment]
Well done! It works properly with the newest TB. 
I will test it more and better on weekdays when I receive dozens of emails.

File: mark-gmail-read/bootstrap.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-ChromeUtils.import("resource:///modules/iteratorUtils.jsm");
-ChromeUtils.import("resource:///modules/mailServices.js");
-ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
+Components.utils.import("resource:///modules/iteratorUtils.jsm");
+ChromeUtils.import("resource:///modules/MailServices.jsm");
+const { XPCOMUtils } = ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
```
Comment (acabal) [inline mark-gmail-read/bootstrap.js:5]: What does this line do? It looks like we're assigning a const, but it's never used anywhere?

File: mark-gmail-read/bootstrap.js [hunk 2]
```diff
@@ -100,6 +100,8 @@ function startup(data, reason){
 	markGmailRead.MSG_FOLDER_GMAIL_IMPORTANT = prefs.getStringPref("folder_names.important", markGmailRead.MSG_FOLDER_GMAIL_IMPORTANT);
 	markGmailRead.MSG_FOLDER_GMAIL_STARRED = prefs.getStringPref("folder_names.starred", markGmailRead.MSG_FOLDER_GMAIL_STARRED);
 
+	console.log("Mark GMail Read: Addon started. Not marking directories '" + markGmailRead.MSG_FOLDER_GMAIL_ALL_MAIL + "', '" + markGmailRead.MSG_FOLDER_GMAIL_IMPORTANT + "' and '" + markGmailRead.MSG_FOLDER_GMAIL_STARRED + "' read, because that would mess with the emails in other folders. Please, make sure these names correspond to the names you really see in the folder pane under the [GMail] folder, otherwise bad things can happen. Set these names as extensions.mark_gmail_read.(all_mail|important|starred).");
```
Comment (acabal) [inline mark-gmail-read/bootstrap.js]: I would prefer it if we don't output to the console with non-error messages. Well-behaved programs are quiet on success. Instead you can put this kind of informational stuff in the installation instructions (which it looks like you did).

**Comment by acabal** [comment]
Thanks! I've left some notes. Let me know if an updated version appears to be working OK after a week or so and I can release this to the TB extensions site. (I don't use Gmail any more so I can't test this myself.)

**Comment by tooomm** [comment]
@acabal No notes visible here.

Edit: Now it works 👍

**Review by acabal — CHANGES_REQUESTED** [review]

File: mark-gmail-read/bootstrap.js (hunk 1 above)
Comment (peci1) [reply mark-gmail-read/bootstrap.js:5]: It is used: XPCOMUtils.defineLazyServiceGetter in startup()

File: mark-gmail-read/bootstrap.js (hunk 2 above)
Comment (peci1) [reply mark-gmail-read/bootstrap.js]: Normally, I wouldn't output. But here, the console message contains the "resolved" config variables, so the user can make sure he had set the correct ones, which is important to know.

Comment (acabal) [reply mark-gmail-read/bootstrap.js]: Yes but if they're not going to read the README, they're not going to read the console output either.; I would rather keep it silent. And, in any case, outputting to the console has nothing to do with TB 68 compatibility.

Comment (peci1) [reply mark-gmail-read/bootstrap.js]: Okay, never mind.

**Comment by peci1** [comment]
Updated

**Comment by acabal** [comment]
Great, thanks!

**Pull request merged**